### PR TITLE
Initial web app: load recipe information automatically

### PR DESCRIPTION
**Pull request by TrangEmily**
This PR solve issue #49 
This is the initial web app which load recipes information automatically and run successfully some recipes. The webpage has 3 columns: 
* List of recipe names 
* Recipe information and user data input: `get_docstrings()`, `get_recipe_contexts()` function
* Result of running recipe

### CODE REVIEW

File: adr/recipe.py [hunk 1]
```diff
@@ -120,3 +131,9 @@ def run_recipe(recipe, args, config):
 
     log.debug("Result:")
     return fmt(output)
+
+
+def get_docstring(recipe):
+    modname = '.recipes.{}'.format(recipe)
+    mod = importlib.import_module(modname, package='adr')
+    return markdown.markdown(mod.__doc__)
```
[the remark this answers is not in the document]
Comment (TrangEmily) [reply adr/recipe.py]: Thanks, updated.

File: adr/recipe.py [hunk 2]
```diff
@@ -94,19 +89,35 @@ def run_recipe(recipe, args, config):
...
+
+    """
     set_config(config)
 
+    modname = '.recipes.{}'.format(recipe)
+    mod = importlib.import_module(modname, package='adr')
+
+    query_context_def, run_context_def = get_recipe_contexts(recipe)
+    recipe_context_def = {**query_context_def, **run_context_def}
+    parsed_args = cli_args_transform(recipe_context_def, args) if from_cli else args
```
[the remark this answers is not in the document]
Comment (TrangEmily) [reply adr/recipe.py]: Thanks, I removed the `cli_args_transform` function

**Comment by TrangEmily** [comment]
> The search bar doesn't seem to work, is this planned for a follow-up? Also if the reset button at the bottom of the task list is meant to clear the search, then maybe it would be better if it were next to the search input box.
Yes the search bar doesn't work now. I will update it. I also replaced the reset button by "x" inside the search box.
With current push, the recipe_contexts is parsed instead of query_contexts and run_contexts, because it causes error in some recipes. For example, query file has no `limit` or `sort-key` context, but (in old version) the recipe calls the query with these contexts, so if we use query_contexts and run_contexts separately it will not operate the same way as old version.

File: adr/context.py [hunk 3]
```diff
@@ -4,12 +4,13 @@
...
-                  'default': None,
-                  'help': "Display values of specified attribute within --table."},
+    'attribute': [['--at'],
+                  {'nargs': '?',
+                      'default': None,
+                      'help': "Display values of specified attribute within --table."}],
     'branch': [['-B', '--branch'],
                {'default': 'mozilla-central',
-                'help': "Branches to query results from",
+                'help': "Btributeranches to query results from",
```
[the remark this answers is not in the document]
Comment (TrangEmily) [reply adr/context.py]: thanks I updated

File: adr/recipe.py [hunk 4]
```diff
@@ -113,16 +110,21 @@ def run_recipe(recipe, args, config, from_cli=True):
...
-    query_args = {k: v for k, v in parsed_args.items() if k in query_context_def}
-    run_args = {k: v for k, v in parsed_args.items() if k in run_context_def}
+    if from_cli:
+        parsed_args = vars(RecipeParser(recipe_context_def).parse_args(args))
+    else:
+        parsed_args = args
+
+    # # Split recipe_args into query_args and recipe_args while keep --help works correctly
+    # query_args = {k: v for k, v in parsed_args.items() if k in query_context_def}
+    # run_args = {k: v for k, v in parsed_args.items() if k in run_context_def}
```
[the remark this answers is not in the document]
Comment (TrangEmily) [reply adr/recipe.py]: thanks I removed it

File: setup.py [hunk 5]
```diff
@@ -16,7 +16,9 @@
     'pyyaml',
     'beautifulsoup4',
     'flask',
-    'markdown'
+    'markdown',
```
[the remark this answers is not in the document]
Comment (TrangEmily) [reply setup.py]: Yes I use it to format the result of recipes

**Comment by ahal** [comment]
Thanks for the fixes!

**Pull request merged**